src/base/cubstream.hpp

```diff
@@ -21,22 +21,18 @@
  * cubstream.hpp
  */
 
-#ifndef _CUBSTREAM_HPP_
-#define _CUBSTREAM_HPP_
-
 #ident "$Id$"
 
-#include <vector>
-#include <functional>
-#include <atomic>
+#ifndef _CUBSTREAM_HPP_
+#define _CUBSTREAM_HPP_
 
-/* TODO[arnia] these will be added from razvan's branch repl_base */
-typedef unsigned long long stream_position;
-typedef char BUFFER_UNIT;
+#include <stddef.h>
 
 namespace cubstream
 {
 
+  typedef unsigned long long stream_position;
```

Comment:
@razvanradulescu 

What about using `off_t` instead of `unsigned long long`?

Reply:
Using off_t results in runtime issues due to multiple definitions of off_t in more than one header. I preferred to use std::uint64_t.